### ISSUE TITLE
the values shown in variable explorere are incorerct

### ISSUE THREAD

**Issue by devineniraghava** [spyder-ide/spyder#13932]
### What steps will reproduce the problem?

I import a dataframe from sql and check my values in variable explorer. The values shown do not match with the values in my database. I tried to reproduce the same with older version of spyder and it works perfectly fine

## Versions

* Spyder version: 4.1.4 
* Python version: 3.7.6
* Qt version: 5.12.5
* PyQt5 version: 5.12.3
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0           :  1.4.0 (OK)
chardet >=2.0.0                :  3.0.4 (OK)
cloudpickle >=0.5.0            :  1.6.0 (OK)
diff_match_patch >=20181111    :  20200713 (OK)
intervaltree                   :  None (OK)
IPython >=4.0                  :  7.18.1 (OK)
jedi =0.17.1                   :  0.17.1 (OK)
nbconvert >=4.0                :  6.0.7 (OK)
numpydoc >=0.6.0               :  1.1.0 (OK)
paramiko >=2.4.0               :  2.7.2 (OK)
parso =0.7.0                   :  0.7.0 (OK)
pexpect >=4.4.0                :  4.8.0 (OK)
pickleshare >=0.4              :  0.7.5 (OK)
psutil >=5.3                   :  5.7.2 (OK)
pygments >=2.0                 :  2.7.1 (OK)
pylint >=1.0                   :  2.6.0 (OK)
pyls >=0.34.0;<1.0.0           :  0.35.1 (OK)
qdarkstyle >=2.8               :  2.8.1 (OK)
qtawesome >=0.5.7              :  1.0.1 (OK)
qtconsole >=4.6.0              :  4.7.7 (OK)
qtpy >=1.5.0                   :  1.9.0 (OK)
rtree >=0.8.3                  :  0.9.4 (OK)
sphinx >=0.6.6                 :  3.2.1 (OK)
spyder_kernels >=1.9.2;<1.10.0 :  1.9.3 (OK)
watchdog                       :  None (OK)
zmq >=17                       :  19.0.2 (OK)

# Optional:
cython >=0.21                  :  None (OK)
matplotlib >=2.0.0             :  3.3.2 (OK)
numpy >=1.7                    :  1.18.1 (OK)
pandas >=0.13.1                :  1.1.2 (OK)
scipy >=0.17.0                 :  1.4.1 (OK)
sympy >=0.7.3                  :  None (OK)
```

**Comment by steff456** [comment]
Hi @devineniraghava,

Can you please upload a gif with the behavior? And also can you give me a small explanation of your setup so we can try to reproduce this?

**Comment by kaleb-keny** [comment]
Yeah I believe I am getting the same thing, at certain times the variables just hang.... Although I restart the kernel the old variables still show in the variables explorer pane....

**Comment by ccordoba12** [comment]
@kaleb-keny, please update to our latest version to see if your problem is fixed on it.

For that please close Spyder, open the Anaconda prompt and run there

    conda update anaconda
    conda install spyder=4.1.5

**Comment by kaleb-keny** [comment]
Hey @ccordoba12 I believe you were right... I was surprised honestly as I just reinstalled both anaconda and spyder, maybe old versions were available from the site... 
Anyway, the issue is resolved...
Just wanted to mention if you have different env... probably people need to run
`conda update -n myenv --all`

**Comment by kaleb-keny** [comment]
Alright I guess i just did the above updates and the issue persists... I believe I might have an idea about what causing this... if you have an async program running and I interrupt the program with a restart kernel... The variable in variable explorer doesn't get reset... However if I do an keyboard interrupt and I kill the tasks ... and then restart kernel... The variables disappear

**Comment by ccordoba12** [comment]
I couldn't reproduce your problem on Linux. Could you upload an animated gif (using the Licecap program) to better see how to replicate the problem.

Also, please don't forget to share your Python version too.

**Comment by kaleb-keny** [comment]
Sure I'll do it now give me few minutes 
* Spyder version: 4.1.5 None
* Python version: 3.8.5 64-bit
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

**Comment by kaleb-keny** [comment]
![test](https://user-images.githubusercontent.com/62237347/95658827-a4b1b980-0b1d-11eb-90b6-4470688f4418.gif)

**Comment by ccordoba12** [comment]
Hey @kaleb-keny, this problem should be fixed in our latest version available in Anaconda (5.3.3). Please update by closing Spyder, opening the Anaconda Prompt and running there the commands mentioned in [our documentation](http://docs.spyder-ide.org/current/installation.html#updating-with-anaconda).
